Src/bat_charge.c

```diff
@@ -0,0 +1,96 @@
+/*
+ * bat_charge.c
+ *
+ *  Created on: Sep, 26 2020
+ *      Author: Luke Oxley
+ */
+#include "bat_charge.h"
+
+uint8_t temp_array[READ_MSG_SIZE];
+uint8_t read_byte = 0;
+uint8_t write_data[WRITE_MSG_SIZE + 1];
+
+void init_BQ21062(uint8_t * write_data);
+
+void task_manage_charger() 
+{
+    uint8_t currentF, voltageF, tempF;
+
+    while(1)
+    {
+        read_byte = set_address(STAT0_ADDR, READ_ENABLE);
+        HAL_I2C_Master_Receive(&hi2c1, read_byte, &temp_array[0], READ_MSG_SIZE, 0xFFFF);
+        while (hi2c1.State != HAL_I2C_STATE_READY)
+	    {
+	        //Wait for the send to stop
+	    }
+        currentF = get_bit(temp_array[0], CURRENT_LIM_BIT);
+
+        read_byte = set_address(STAT1_ADDR, READ_ENABLE);
+        HAL_I2C_Master_Receive(&hi2c1, read_byte, &temp_array[0], READ_MSG_SIZE, 0xFFFF);
+        while (hi2c1.State != HAL_I2C_STATE_READY)
+	    {
+	        //Wait for the send to stop
+	    }
+        voltageF = get_bit(temp_array[0], OVERVOLTAGE_BIT);
+        tempF = get_bit(temp_array[0], HOT_BIT);
+
+        //TODO Report faults to something else
+        
+        vTaskDelay(200);
+    }
+
+}
+
+void setChargeEnable(uint8_t enable)
+{
+    write_data[0] = set_address(ICCTRL2_ADDR, WRITE_ENABLE);
+  
+    if(enable) write_data[1] = ICCTRL2_DEF || CHRG_ENABLE;
+    else write_data[1] = ICCTRL2_DEF || CHRG_DISABLE;
+    HAL_I2C_Master_Transmit(&hi2c1, write_data[0], &write_data[1], WRITE_MSG_SIZE, 0xFFFF);
+
+    while (hi2c1.State != HAL_I2C_STATE_READY)
+	{
+	  //Wait for the send to stop
+	}
+    return;
```

Comment:
unnecessary return for a void function.
Nothing wrong with it, you just don't need it since it is at the end of the function. If you were trying to manage control flow, for instance in the case of an error, then you can return before the end of a void function. However, if there is no control flow and the function executes from start to finish, then this return is extraneous.